### PR TITLE
Support PD Etcd Client

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

Ref https://github.com/pingcap/tidb-operator/issues/2331

To support Dashboard query metric ability, we should deliver the monitor information (host/port) to PD etcd.

This request add etcdClient in PDClient in order to Put/Delete Key/Value in PD Etcd.

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/pdapi/pdetcd.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	return &pdEtcdClient{
+		etcdClient: etcdClient,
+		timeout:    timeout,
+	}, nil
+}
+
+func (pec *pdEtcdClient) PutKey(key, value string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), pec.timeout)
+	defer cancel()
+	_, err := pec.etcdClient.Put(ctx, key, value)
```
Comment (weekface) [inline pkg/pdapi/pdetcd.go:53]:
```suggestion
	return pec.etcdClient.Put(ctx, key, value)
```

why not return the original result?

Comment (Yisaer) [reply pkg/pdapi/pdetcd.go:53]: I think we don't need to handle the etcd response.  we only need to ensure the `PutKey` is success by check whether error is nil.

File: pkg/pdapi/pdetcd.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	DeleteKey(key string) error
+}
+
+type pdEtcdClient struct {
+	timeout    time.Duration
+	etcdClient *etcdclientv3.Client
+}
+
+func NewPdEtcdClient(url string, timeout time.Duration, tlsConfig *tls.Config) (PDEtcdApi, error) {
```
Comment (DanielZhangQD) [inline pkg/pdapi/pdetcd.go:35]: Why do we need etcd client? Can we use the pdclient?

Comment (Yisaer) [reply pkg/pdapi/pdetcd.go:35]: We use etcdClient to set etcd key in PD in order to support dashboard.

**Pull request merged**